flake.nix

```diff
@@ -0,0 +1,32 @@
+{
+  description = "AWS-LC is a general-purpose cryptographic library";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.11";
+
+  outputs = { self, nix, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        shells = import ./nix/devshell.nix { pkgs = pkgs; };
+
+      in rec {
+        packages.aws-lc = pkgs.stdenv.mkDerivation {
+          src = self;
+          name = "aws-lc";
+          inherit system;
+          nativeBuildInputs = [ pkgs.ninja pkgs.cmake pkgs.perl pkgs.golang ];
```

Comment:
What do these dependencies do exactly? Should we try and minimize them? Such as no go and perl?

Reply:
These are needed for the build.  My original attempt at [this flake](https://github.com/dougch/aws-lc/blob/nixv1.36.0/flake.nix#L14) didn't have Go, but had minimal testing.  I secretly have a fips version of this flake that needs go, so I'm including it here for future-proofing.